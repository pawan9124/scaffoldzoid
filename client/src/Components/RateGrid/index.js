import React, { useState } from "react";
import EditIcon from "@material-ui/icons/Edit";
import DeleteIcon from "@material-ui/icons/Delete";
import DataGrid from "react-data-grid";
import ManageOrange from "../MangeOrange";
import { deleteRate } from "../../actions/rateChartActions";
import { useDispatch, useSelector } from "react-redux";
import { withRouter } from "react-router-dom";

const RateGrid = (props) => {
  const [isOpen, setIsOpen] = useState(false);
  const [editData, setEditData] = useState({});

  const dispatchProps = useDispatch();
  const userDetails = useSelector((state) => state.auth.user);

  /* Format the cell with the icon */
  const actionCellFormatter = ({ value, row }) => {
    return (
      <div className="chart_action_buttons">
        <EditIcon titleAccess="Edit" onClick={(e) => editRow(e, value, row)} />
        <DeleteIcon
          titleAccess="Delete"
          onClick={(e) => deleteRow(e, value, row)}
        />
      </div>
    );
  };

  const columns = [
    { key: "type", name: "Orange Type", width: "50%" },
    {
      key: "rate",
      name: "Orange Rate (Rs/Kg)",
      width:
        userDetails?.isSeller && props.match.path.indexOf("profile") === -1
          ? "30%"
          : "50%",
    },
  ];

  if (userDetails?.isSeller && props.match.path.indexOf("profile") === -1) {
    columns.push({
      key: "action",
      name: "Action",
      formatter: actionCellFormatter,
      width: "20%",
    });
  }

  const editRow = (e, i, data) => {
    setIsOpen(true);
    setEditData(data);
  };
  const deleteRow = (e, i, data) => {
    dispatchProps(deleteRate(data._id, userDetails.id));
  };
  return (
    <div data-test="RateGridComponent">
      <DataGrid
        columns={columns}
        rows={props.rows}
        rowsCount={3}
        enableRowSelect={false}
      />
      {isOpen && (
        <ManageOrange
          isOpen={isOpen}
          setIsOpen={setIsOpen}
          eventType={"edit"}
          editData={editData}
        />
      )}
    </div>
  );
};

export default withRouter(RateGrid);
