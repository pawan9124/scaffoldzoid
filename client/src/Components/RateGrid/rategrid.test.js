import React from "react";
import { mount } from "enzyme";
import { findByTestAttr, checkProps, createMockStore } from "../../testUtils";
import RateGrid from "./index";
import renderer from "react-test-renderer";
import { withRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";

const props = {
  history: {},
  location: {},
  match: {},
  staticContext: undefined,
  rows: [],
};
const store = createMockStore({
  auth: { isAuthenticated: true, user: {} },
  errors: {},
  profile: { profiles: {}, allProfiles: [] },
  rates: { currentRate: {}, allRates: [] },
});

describe("Taking snap shot", () => {
  it("Taking the snapshot of the Component", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <RateGrid {...props} />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Checking the render of component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <RateGrid {...props} />
        </Router>
      </Provider>
    );
  });
  it("Should Render a component", () => {
    const component = findByTestAttr(wrapper, "RateGridComponent");
    expect(component.length).toBe(1);
  });
});

describe("Checking PropTypes", () => {
  it("Should Not throw a warning", () => {
    const expectedProps = props;

    const propsError = checkProps(RateGrid, expectedProps);
    expect(propsError).toBeUndefined();
  });
});
