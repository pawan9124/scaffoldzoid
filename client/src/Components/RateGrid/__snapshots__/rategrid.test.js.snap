// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taking snap shot Taking the snapshot of the Component 1`] = `
<div
  data-test="RateGridComponent"
>
  <div
    aria-colcount={2}
    aria-rowcount={1}
    className="rdg r104f42s700-canary36"
    onScroll={[Function]}
    role="grid"
    style={
      Object {
        "--filter-row-height": "45px",
        "--header-row-height": "35px",
        "--row-height": "35px",
        "--row-width": "160px",
        "--template-columns": "80px 80px ",
      }
    }
  >
    <div
      aria-rowindex={1}
      className="rdg-header-row h1fquj5h700-canary36 h19m9p7h700-canary36"
      role="row"
    >
      <div
        aria-colindex={1}
        className="rdg-cell c1wupbe700-canary36"
        role="columnheader"
        style={
          Object {
            "gridColumnStart": 1,
          }
        }
      >
        Orange Type
      </div>
      <div
        aria-colindex={2}
        className="rdg-cell c1wupbe700-canary36"
        role="columnheader"
        style={
          Object {
            "gridColumnStart": 2,
          }
        }
      >
        Orange Rate (Rs/Kg)
      </div>
    </div>
    <div
      className="rdg-focus-sink f7ly7s700-canary36"
      onFocus={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    />
    <div
      style={
        Object {
          "height": 0,
        }
      }
    />
  </div>
</div>
`;
